### PR TITLE
Add aws sno etcd encryption test profiles

### PR DESCRIPTION
**Pull request by wangke19**
The following changes have been made：
1. Optimize the etcd encryption, especially SNO
2. Add `cucushift-installer-rehearse-aws-ipi-sno-etcd-encryption` workflow
3. Add aws SNO test profiles(4.12- 4.16)

### CODE REVIEW

**Comment by wangke19** [comment]
/pj-rehearse  periodic-ci-openshift-openshift-tests-private-release-4.12-multi-nightly-azure-ipi-ovn-etcd-encryption-arm-f28

**Comment by wangke19** [comment]
/pj-rehearse  periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-sno-etcd-encryption-f7

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-etcd-encryption-f14

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-aws-ipi-sno-etcd-encryption-f14

**Comment by wangke19** [comment]
pj-rehearse pull-ci-openshift-gcp-pd-csi-driver-operator-release-4.16-e2e-gcp-csi-extended

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-vsphere-ipi-compact-etcd-encryption-f14

**Comment by wangke19** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-vsphere-ipi-compact-etcd-encryption-f14

**Comment by wangke19** [comment]
etcd-encryption succeeded on the below test jobs, 
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-sno-etcd-encryption-f14
```
INFO[2024-03-11T11:50:50Z] Running step aws-ipi-sno-etcd-encryption-f14-etcd-encryption. 
INFO[2024-03-11T11:59:48Z] Step aws-ipi-sno-etcd-encryption-f14-etcd-encryption succeeded after 8m58s. 
```

ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-vsphere-ipi-compact-etcd-encryption-f14
```
INFO[2024-03-11T17:10:25Z] Running step vsphere-ipi-compact-etcd-encryption-f14-etcd-encryption. 
INFO[2024-03-11T17:32:01Z] Step vsphere-ipi-compact-etcd-encryption-f14-etcd-encryption succeeded after 21m35s. 
```
ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-sno-etcd-encryption-f7
```
INFO[2024-03-11T11:47:25Z] Running step aws-ipi-sno-etcd-encryption-f7-etcd-encryption. 
INFO[2024-03-11T11:56:24Z] Step aws-ipi-sno-etcd-encryption-f7-etcd-encryption succeeded after 8m58s. 
```

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml [hunk 1]
```diff
@@ -660,6 +660,16 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-aws-usgov-ipi-private-workers-marketplace
+- as: aws-ipi-sno-etcd-encryption-f14
+  cron: 5 18 4,18 * *
+  steps:
+    cluster_profile: aws-qe
+    env:
+      BASE_DOMAIN: qe.devcluster.openshift.com
+      TEST_FILTERS: ~ChkUpgrade&;~NonPreRelease&;~Disruptive&;~Serial&;~DisconnectedOnly&;~HyperShiftMGMT&;~MicroShiftOnly&
```
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml:669]: `TEST_FILTERS` is not needed.  Most of the time, we overwrite it for disconnected clusters.

Comment (wangke19) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml]: Updated.

File: ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15-periodics.yaml [hunk 2]
```diff
@@ -9174,17 +9174,6 @@ periodics:
     job-release: "4.15"
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
   name: periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-f28-longduration-hive-remaining
-  reporter_config:
-    slack:
-      channel: '#forum-prow-hive-qe-ci'
```
Comment (liangxia) [inline ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15-periodics.yaml:9179]: Those removal are unexpected.

Comment (wangke19) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15-periodics.yaml:9179]: It's not from me, I ran `make update`, these contents are removed.

Comment (wangke19) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15-periodics.yaml]: Get it fixed manually.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml [hunk 3]
```diff
@@ -690,6 +690,15 @@ tests:
     test:
     - chain: openshift-e2e-test-qe-destructive
     workflow: cucushift-installer-rehearse-aws-usgov-ipi-private-workers-marketplace
+- as: aws-ipi-sno-etcd-encryption-f14
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:693]: we want to treat this job as a normal ci job, right? If you just want to cover sno + etcd-encryption case, we already have `aws-ipi-disc-priv-sno` job in multi-nightly.yaml. You can append `etcd-encryption` step to create a new install workflow `aws-ipi-disc-priv-sno-etcd-encryption`, or even, you can directly replace `aws-ipi-disc-priv-sno` with `aws-ipi-sno-etcd-encryption`

**Comment by jianlinliu** [comment]
@wangke19 sorry to be late to review, I just left some comments.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml (hunk 3 above)
Comment (wangke19) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:693]: The workflow ` workflow: cucushift-installer-rehearse-aws-ipi-sno-etcd-encryption` is used for another recert job https://github.com/openshift/release/pull/49183. To add these sno + etcd-encryption test here, I can consider merge them into multi-nightly.yaml in another job.

Comment (wangke19) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml:693]: Raised another pr https://github.com/openshift/release/pull/49845 for your suggestion.

**Pull request merged**